cupy/core/core.pyx

```diff
@@ -1842,7 +1842,7 @@ cpdef str _get_header_source():
     if _header_source is None:
         source = []
         base_path = _get_header_dir_path()
-        for file_path in _cupy_header_list + _cupy_extra_header_list:
+        for file_path in _cupy_extra_header_list + _cupy_header_list:
```

Comment:
Just out of curiosity, why needs the order need to be reversed?

Reply:
Because we are now handling the header include manually for hiprtc, all headers must appear in order. This is essentially what a functional compiler would do for us (the first entry cupy/complex/complex.h in "extra headers" includes all of the declarations and must go first).